ovirtlago/virt.py

```diff
@@ -118,34 +123,51 @@ def wait_for_ssh(self):
 class EngineVM(lago.vm.DefaultVM):
     def __init__(self, *args, **kwargs):
         super(EngineVM, self).__init__(*args, **kwargs)
-        self._api = None
+        self._api_v3 = None
+        self._api_v4 = None
 
     def stop(self):
         super(EngineVM, self).stop()
-        self._api = None
+        self._api_v3 = None
 
     def _artifact_paths(self):
         inherited_artifacts = super(EngineVM, self)._artifact_paths()
         return set(inherited_artifacts + ['/var/log/ovirt-engine', ])
 
-    def _create_api(self):
+    def _create_api(self, api_ver):
         url = 'https://%s/ovirt-engine/api' % self.ip()
-        return ovirtsdk.api.API(
-            url=url,
-            username=constants.ENGINE_USER,
-            password=str(self.metadata['ovirt-engine-password']),
-            validate_cert_chain=False,
-            insecure=True,
-        )
+        if api_ver == 3:
+            return ovirtsdk.api.API(
+                url=url,
+                username=constants.ENGINE_USER,
+                password=str(self.metadata['ovirt-engine-password']),
+                validate_cert_chain=False,
+                insecure=True,
+            )
+        if api_ver == 4:
+            if not API_V4:
+                raise RuntimeError('oVirt Python SDK v4 not found.')
+            return sdk4.Connection(
+                url=url,
+                username=constants.ENGINE_USER,
+                password=str(self.metadata['ovirt-engine-password']),
+                insecure=True,
+                debug=True,
+            )
+        raise RuntimeError('Unknown API requested: %s' % api_ver)
 
-    def _get_api(self):
+    def _get_api(self, api_ver):
         try:
-            api = []
+            api_v3 = []
+            api_v4 = []
 
             def get():
```

Comment:
We are using the closure because ```testlib.assert_true_within``` isn't returning the return value of the function that it runs.

I think that we should keep ```testlib.assert_true_within``` the same it is now, as it allow us to separate between the return value of the inner function and the result of the timeout assertion (for example a case where the assertion failed but we still want to get the return value of the inner function).